src/Models/Artificial Intelligence/KongAI.js

```diff
@@ -13,30 +14,114 @@ export default class KongAI extends Player{
         }
         this.lookAheadDepth = 2;
         this.boardStatus = gameBoard;
-        console.log("AI Active");
+        this.KingPawn = "Lion";
 
 
     } 
     updateBoard(gameBoard){
         this.boardStatus = gameBoard;
 
     }
+    identifyDifference(prevStatus,currStatus){
+        for(let i=0;i<prevStatus.length;i++){
+            if(prevStatus[i].Position.x != currStatus[i].Position.x && 
+                prevStatus[i].Position.y != currStatus[i].Position.y){
+                    console.log(prevStatus[i].Position,currStatus[i].Position);
+                    return true;
+                }
+        }
+    }
+    findOptimalState(){
+        let searchSpace = this.GetActionStates();
+        // Apply Score based on minimax 
+        this.applyMiniMaxScore(searchSpace,true);
+        
+        let indicesOfMax = [];
+        let currMax = searchSpace["score"];
+        // Locate the Optimal Board State
+        searchSpace = searchSpace["children"];
+        for(let i = 0; i < searchSpace.length; i++){
+            if(searchSpace[i]["score"] == currMax){
+                indicesOfMax.push(i);
+            }else if(searchSpace[i]["score"] > currMax){
+                indicesOfMax = [];
+                indicesOfMax.push(i);
+                currMax = searchSpace[i]["score"];
+            }
+            if (this.identifyDifference(this.boardStatus.PlayerTwo.ActivePawns,searchSpace[i]["board"].PlayerTwo.ActivePawns)){
+                return searchSpace[i]["board"];
+            }
+        }
+        // this.identifyDifference(this.boardStatus.PlayerTwo.ActivePawns,searchSpace[indicesOfMax[Math.floor(Math.random()*(indicesOfMax.length-1))]]["board"].PlayerTwo.ActivePawns);
+        return searchSpace[indicesOfMax[Math.floor(Math.random()*(indicesOfMax.length-1))]]["board"];
+        
+    }
+    applyMiniMaxScore(root,toMaximize){
+        if(!root["children"])
+            return;
+
+        if(toMaximize){
+            for(let state of root["children"]){
+                this.applyMiniMaxScore(state,false);
+                // this.identifyDifference(this.boardStatus.PlayerTwo.ActivePawns,state["board"].PlayerTwo.ActivePawns);
+                if(state["children"])
+                    root["score"] = this.getMaxScore(state["children"]);
+            }
+                
+        }
+        else{
+            for(let state of root["children"]){
+                this.applyMiniMaxScore(state,true);
+                // this.identifyDifference(this.boardStatus.PlayerTwo.ActivePawns,state["board"].PlayerTwo.ActivePawns);
+                if(state["children"])
+                    root["score"] = this.getMinScore(state["children"]);
+            }
+        }
+        return;
+        
+    }
+    getMaxScore(childrenStates){
 
-    performAction(){
+        let currMax = childrenStates[0]["score"];
+        
+        // Locate the Optimal Board State
+        for(let i = 0; i < childrenStates.length; i++){
```

Comment:
Since currMax was initialized using the first state, shouldn't the loop be 
`for(let i = 1; i < childrenStates.length - 1; i++){`?

Reply:
It's not a huge issue. I would say it would still function the same way.